### ISSUE TITLE
Integrating with gRPC

### ISSUE THREAD

**Issue by MyPierre** [simpleinjector/SimpleInjector#709]
Hi,
I have some difficulties to see how to configure a container for the gRPC framework.
Please could advise is there anyway/example to implement it ?
Thank you for your help.

**Comment by dotnetjunkie** [comment]
I'm unfamiliar with the gRPC framework. Can you show me what you tried and what examples you found for other DI Containers?

**Comment by MyPierre** [comment]
I found a discussion related to DI and gRPC proposing an idea here :
https://github.com/grpc/grpc/issues/12478
the example on which it is based to develop a solution idea comes from the grpc documentation: https://grpc.io/docs/quickstart/csharp/
I don't see what is a correct way to do it using simplinjector.
If you have an idea, it will be valuable to me.

**Comment by dotnetjunkie** [comment]
The solution with Simple Injector would be similar to the example given [here](https://github.com/grpc/grpc/issues/12478#issuecomment-328555173). In other words, you turn your gRPC class into a [Humble Object](http://xunitpatterns.com/Humble%20Object.html):

``` c#
class GreeterImpl : Greeter.GreeterBase
{
    private Container container;

    public GreeterImpl(Container container) => this.container = container;

    public override async Task<HelloReply> SayHello(HelloRequest request, ServerCallContext context)
    {
        using (AsyncScopedLifestyle.BeginScope(this.container))
        {
            // NOTE: Moved all logic to the GreeterService implementation
            var service = container.GetInstance<GreeterService>();
            return await service.SayHello();
        }
    }
}
```

``` c#
Server server = new Server
{
    Services = { Greeter.BindService(new GreeterImpl(container)) },
}
```

**Comment by MyPierre** [comment]
Thank you dotnetjunkie  for this enlightening solution but also the fast response time.

**Comment by DanielLebon** [comment]
Hi,
I am just starting to use  simpleinjecor and Grpc implementation is an Interesting topics for me.
I have a question for dotnetjunkie , how this service can be register in the container ?
Thank you  for enlightening me on this.

**Comment by dotnetjunkie** [comment]
> how this service can be register in the container ?

Can you rephrase your question?

**Comment by DanielLebon** [comment]
Hello dotnetjunkie,
Thank you for the answer.
My Bad, after analyzing the solution, I understood the error of my question.
I'll delete it, it'll be cleaner.

**Comment by dotnetjunkie** [comment]
Hi @MyPierre, after [discussing](https://github.com/grpc/grpc-dotnet/issues/671) the addition of new interception points to gRPC, Microsoft recently released a pre-release of 2.26 that now allows Simple Injector to integrate with gRPC.

You can read [here](https://github.com/simpleinjector/SimpleInjector/issues/785) how to integrate.

**Comment by MyPierre** [comment]
Thank you for this interesting update.
I have already tested it and it seems to work perfectly.
Thanks again.

**Comment by DanielLebon** [comment]
Indeed, nice improvement, I also tested it because it actually simplifies the code.
Just for additional information, we  also have to add the package SimpleInjector.Integration.AspNetCore.Mvc.   (https://simpleinjector.readthedocs.io/en/latest/aspnetintegration.html)
